shared/AppInsightsCore/src/JavaScriptSDK/CoreUtils.ts

```diff
@@ -1,379 +1,223 @@
- * @param target - The target object to be assigned with the source properties and functions
- * @param source - The source object which will be assigned / called by setting / calling the targets proxies
- * @param chkSet - An optional callback to determine whether a specific property/function should be proxied
- * @memberof Initialization
+ * Provides a collection of utility functions, included for backward compatibility with previous releases.
+ * @deprecated Marking this interface and instance as deprecated in favor of direct usage of the helper functions
+ * as direct usage provides better tree-shaking and minification by avoiding the inclusion of the unused items 
+ * in your resulting code.
  */
-export function proxyAssign(target: any, source: any, chkSet?: (name: string, isFunc?: boolean, source?: any, target?: any) => boolean) {
-    if (target && source && target !== source && _isObject(target) && _isObject(source)) {
-        // effectively apply/proxy full source to the target instance
-        for (const field in source) {
-            if (CoreUtils.isString(field)) {
-                let value = source[field] as any;
-                if (_isFunction(value)) {
-                    if (!chkSet || chkSet(field, true, source, target)) {
-                        // Create a proxy function rather than just copying the (possible) prototype to the new object as an instance function
-                        target[field as string] = (function(funcName: string) {
-                            return function() {
-                                // Capture the original arguments passed to the method
-                                var originalArguments = arguments;
-                                return source[funcName].apply(source, originalArguments);
-                            }
-                        })(field);
-                    }
-                } else if (!chkSet || chkSet(field, false, source, target)) {
-                    if (_hasOwnProperty(target, field)) {
-                        // Remove any previous instance property
-                        delete target[field];
-                    }
-
-                    if (!objDefineAccessors(target, field, () => {
-                        return source[field];
-                    }, (theValue) => {
-                        source[field] = theValue;
-                    })) {
-                        // Unable to create an accessor, so just assign the values as a fallback
-                        // -- this will (mostly) work for objects
-                        // -- but will fail for accessing primitives (if the source changes it) and all types of "setters" as the source won't be modified
-                        target[field as string] = value;
-                    }
-                }
-            }
-        }
-    }
+export interface ICoreUtils {
```

Comment:
Defining an interface for the old "exported" CoreUtils class, so that CoreUtils can become an exported constant that follows the previously exposed class values that just reference the extracted functions -- i.e. CoreUtils should not have any locally defined functions.

There is 1 current exception to this and its the CoreUtils._canUseCookies property -- because it's a property. This is used by the community so it need to stay as a working change -- even when I add my CookieManager changes on top of this.

Reply:
So the changes in this file include
- moving functions from CoreUtils.xxx => first class function (defined and exported directly)
- moving the functions from CoreUtils.xxx => first class function that are now in HelperFuncs.ts and RandomHelper.ts etc
- Defining ICoreUtils interface to match and document the previous CoreUtils
- export a constant CoreUtils: ICoreUtils implementation (start @ line 365)
- Same for EventHelper class